apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java

```diff
@@ -165,6 +165,9 @@ private void handleEvent(ReportingEvent event, long sequence, boolean endOfBatch
             }
             connection = startRequest();
             payloadSerializer.serializeMetaDataNdJson(metaData);
+            if (logger.isTraceEnabled()) {
+                logger.trace(payloadSerializer.toString());
```

Comment:
The payload is already logged when the request ends. What's the benefit of logging the whole payload for every event?

Reply:
Getting it in JSON format, see what we actually send the server

Reply:
But that's already done here: https://github.com/elastic/apm-agent-java/blob/035ba3a2ef30262d6e1983daf36d92128bd76ce2/apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/DslJsonSerializer.java#L111

Reply:
Only if flushed, which never happens if the connection with the server is down: 
https://github.com/elastic/apm-agent-java/blob/035ba3a2ef30262d6e1983daf36d92128bd76ce2/apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java#L282

Reply:
Anyway, not sure the right way to go, I'll remove it anyway and we will find a way to do it properly